doc/developers/contributing.rst

```diff
@@ -251,11 +251,18 @@ If any of the above seems like magic to you, then look up the `Git documentation
 web, or ask a friend or another contributor for help.
 
 If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. The command will be::
+to merge ``master``. For that, you first need to fetch the ``upstream``, and
+then merge its ``master`` into your branch::
 
-  $ git merge master
+  $ git fetch upstream
+  $ git merge upstream/master
 
-with ``master`` being synchronized with the ``upstream``.
+It is recommended to work on temporary branches rather than directly on
```

Comment:
We're talking about how to solve the conflict when merging master from upstream. I can't understand why we mention these things here. (Notice that the next paragraph starts with "Subsequently")

Reply:
Fair point. Should we move it right after where the user clones the fork, and
add the upstream remote etc etc?

Reply:
Just saw that @qinhanmin2014's review is similar to mine :)

Reply:
We already have
```
Create a branch to hold your development changes:
$ git checkout -b my-feature
```
Anyway, this is not a big problem, as long as you can put these things in a suitable place, I won't oppose.